ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfiguration.kt

```diff
@@ -0,0 +1,67 @@
+package com.revenuecat.purchases.ui.revenuecatui.data.processed
+
+import android.net.Uri
+import androidx.compose.ui.graphics.Color
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.PaywallViewMode
+
+internal data class TemplateConfiguration(
+    val template: PaywallTemplate,
+    val mode: PaywallViewMode,
+    val packageConfiguration: PackageConfiguration,
+    val configuration: PaywallData.Configuration,
+    val images: Images,
+) {
+    private val darkModeColors = ColorsFactory.create(configuration.colors, true)
+    private val lightModeColors = ColorsFactory.create(configuration.colors, false)
```

Comment:
Note how I create both sets of colors and the consumer can chose which to use with `getColors`. Did it that way so it's the `Composable` the one in charge of passing the dark theme information, so it's more performant to update to/from dark mode than updating the whole view model.

Reply:
Makes sense 👍🏻 
Just one note in that implementation that would make it simpler to implement.